lib/browser/api/menu.js

```diff
@@ -164,10 +164,15 @@ Menu.buildFromTemplate = function (template) {
   const sorted = sortTemplate(filtered)
 
   sorted.forEach((item) => {
-    if (typeof item !== 'object') {
+    if (typeof item !== ('object' || null)) {
       throw new TypeError('Invalid template for MenuItem')
     }
-    menu.append(new MenuItem(item))
+    if (Object.keys(item).length > 0 && item !== null) {
+      if (item.label === (undefined || null) && item.role === (undefined || null)){
+        throw new TypeError('Invalid template for MenuItem')
+      }
```

Comment:
`(undefined || null)` is evaluted as `null`, so these tests are effectively `item.label === null` and `item.role === null` which is not ideal because we want to check for `undefined` to see if the object has the property.

In this case I think it's reasonable and clearer to use the [truthy](https://developer.mozilla.org/en-US/docs/Glossary/Truthy) test and write this as `if (!item.label && !item.role)`

Reply:
thanks gotcha